connectors/byoc.py

```diff
@@ -578,39 +511,42 @@ def next_sync(self):
         if self.sync_now:
             logger.debug("sync_now is true, syncing!")
             return 0
-        if not self.scheduling["enabled"]:
+        if not self.scheduling.get("enabled", False):
             logger.debug("scheduler is disabled")
             return SYNC_DISABLED
-        return next_run(self.scheduling["interval"])
+        return next_run(self.scheduling.get("interval"))
+
+    async def reset_sync_now_flag(self):
+        await self.index.update(doc_id=self.id, doc={"sync_now": False})
+
+    async def sync_starts(self):
```

Comment:
Why have both `sync_starts` and `_sync_starts`?

Reply:
I want to separate concerns. When you start a sync, you should:
1. claim the job, via `sync_job.claim()`
2. update `last_synced_*` of the connector, via `connector.sync_starts()`.

Currently, there are many logics initiated by the connector, e.g. `connector._sync_starts` will do job creation and claim, and update `last_synced_*`. That's the main purpose of this refactor.

Why have we kept both `sync_starts` and `_sync_starts` now? Because I don't want to make this PR too complicated. Once I refactor `SyncService`, it will 1) create a job via `sync_job_index.create()`, 2) claim the job via `sync_job.claim()`, 3) update `last_synced_*` of the connector via `connector.sync_starts()` (task 2 and 3 will be further refactored to `JobExecutionService` in the future).  

Til then, `_sync_starts` will no longer used and can be removed. The same applies to methods like `sync`, `prepare_docs` of class `Connector`.